runtime/matchers.go

```diff
@@ -0,0 +1,39 @@
+package runtime
+
+import "strings"
+
+// EqualMatcher performs a case-sensitive equality match for request metadata keys
+func EqualMatcher(s string) ServeMuxOption {
```

Comment:
I would prefer a WithHeaderMatcher Option that accepts a ```func(string) bool``` to both support more scenarios but also to have a smaller api surface area.